homeassistant/components/templated_automation/__init__.py

```diff
@@ -0,0 +1,81 @@
+"""
+Support for Templated Automation.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/templated_automation
+"""
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_SOURCE, EVENT_STATE_CHANGED, STATE_OFF, \
+    SERVICE_TURN_OFF, SERVICE_TURN_ON, ATTR_ENTITY_ID, ATTR_STATE
+
+CONF_BIDIRECTIONAL = 'bidirectional'
+CONF_INVERSE = 'inverse'
+CONF_TARGET = 'target'
+
+DEFAULT_BIDIRECTIONAL = False
+DEFAULT_INVERSE = False
+
+DOMAIN = 'templated_automation'
+
+AUTOMATION_TEMPLATE = vol.Schema({
+    vol.Required(CONF_SOURCE): cv.entity_id,
+    vol.Required(CONF_TARGET): cv.entity_id,
+    vol.Optional(
+        CONF_BIDIRECTIONAL, default=DEFAULT_BIDIRECTIONAL
+    ): cv.boolean,
+    vol.Optional(CONF_INVERSE, default=DEFAULT_INVERSE): cv.boolean
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    vol.Optional(DOMAIN): vol.All(cv.ensure_list, [AUTOMATION_TEMPLATE])
```

Comment:
We usually don't explicitly mark the domain as optional.

Reply:
Seen it done both ways, guess that means there's old components that need cleaning up. Will change this one.